salt/states/archive.py

```diff
@@ -1216,6 +1237,26 @@ def extracted(name,
             _add_explanation(ret, source_hash_trigger, contents_missing)
             return ret
 
+        if clean_parent and contents is not None:
+            errors = []
+            log.debug('Removing directory %s due to clean_parent set to True', name)
+            try:
+                salt.utils.files.rm_rf(name.rstrip(os.sep))
+                ret['changes'].setdefault(
+                    'removed', "Directory {} was removed prior to the extraction".format(name))
```

Comment:
Since you've removed the directory, you need to recreate it now. And you also need to take into account the value of the `makedirs` argument when you do. Otherwise, the destination directory won't exist when it comes time to extract, and the extraction will fail.

Reply:
Correct me if I'm wrong but parent directory going to be created no matter what:
```
1286         if not os.path.isdir(name):
1287             __states__['file.directory'](name, user=user, makedirs=True)
1288             created_destdir = True
```

Reply:
Ahh yes, you are correct. I mistakenly thought that this state used a `makedirs` param, I must have been confusing it with something else.